.github/workflows/release.yml

```diff
@@ -30,8 +31,8 @@ jobs:
       - name: Create Release Pull Request or Publish to npm
         uses: changesets/action@v1
         with:
-          publish: yarn run release
-          version: yarn run version
+          publish: pnpm -r publish --access=public
+          version: pnpm run version
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
In order to make the `workspace:` protocol work with publishing, I replaced `changeset publish` with `pnpm -r publish`. `pnpm publish` uses `NODE_AUTH_TOKEN` so I also renamed the `env` var.

`pnpm publish` replaces `workspace:` references with the specific version of the package in the workspace using the semver range specified by the workspace.

For example:

```json
{
	"dependencies": {
		"foo": "workspace:*",
		"bar": "workspace:~",
		"qar": "workspace:^",
		"zoo": "workspace:^1.5.0"
	}
}
```

is transformed into:

```json
{
	"dependencies": {
		"foo": "1.5.0",
		"bar": "~1.5.0",
		"qar": "^1.5.0",
		"zoo": "^1.5.0"
	}
}
```

`pnpm publish` also publishes packages using a dependency graph which guarantees that packages are published in the correct order.

To validate that `pnpm publish` works correctly with `changesets`, I created a repo with dummy packages: https://github.com/with-heart/changesets-pnpm-workspace-protocol-test

[`@pnpm/pnpm`](https://github.com/pnpm/pnpm) itself also uses `changesets` with `pnpm publish`.

Reply:
Good call! Played around with it for a bit to verify that `changeset tag` works. Lemme update our script

Reply:
You can see it working on my test repo: https://github.com/with-heart/changesets-pnpm-workspace-protocol-test